src/Core/Services/MetadataProviders/Converters/ObjectConverter.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.DataApiBuilder.Core.Services.MetadataProviders.Converters
+{
+    /// <summary>
+    /// This is a converter to serialize and deserialize default
+    /// can object can be of different types int, long, string, datetime.
+    /// sample list of types :  SchemaConverter.cs  -> CreateValueNodeFromDbObjectNode
+    /// https://github.com/Azure/data-api-builder/blob/main/src/Service.GraphQLBuilder/Sql/SchemaConverter.cs#L218
+    /// This converter adds a property Type to the object during serialization which stores the type of object and stores the actual value in property Value
+    /// which then gets used during deserialization.
+    /// </summary>
+    public class ObjectConverter : JsonConverter<object>
+    {
+        public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            using (JsonDocument doc = JsonDocument.ParseValue(ref reader))
+            {
+                JsonElement obj = doc.RootElement;
+                string typeName = obj.GetProperty("Type").GetString()!;
+                Type type = Type.GetType(typeName)!;
+                object value = JsonSerializer.Deserialize(obj.GetProperty("Value").GetRawText(), type, options)!;
```

Comment:
now that the code that checked the type and attempted casting was removed, what is the effect? How are the types resolved now?

Reply:
This is the serialized version : ![image](https://github.com/Azure/data-api-builder/assets/82844317/95482d91-d8c7-411d-9091-5bd193b5796f)
deserialized back to '('John')' the original format